pkg/vault/contracts/BasePoolMath.sol

```diff
@@ -238,22 +238,34 @@ library BasePoolMath {
         uint256 invariantRatio = newSupply.divUp(totalSupply);
         ensureInvariantRatioBelowMaximumBound(pool, invariantRatio);
 
-        uint256 newBalance = pool.computeBalance(currentBalances, tokenInIndex, invariantRatio);
+        uint256 amountIn = 0;
+        uint256 fee = 0;
+        {
+            // We remove round up for current balance to make `amountIn` go up.
+            uint256[] memory currentBalancesRoundedDown = new uint256[](currentBalances.length);
+            for (uint256 i = 0; i < currentBalances.length; ++i) {
+                currentBalancesRoundedDown[i] = currentBalances[i] - 1;
+            }
```

Comment:
So, this sounds strictly correct but a bit too much effort for it to be justified. We're allocating a whole new array just to subtract 1 wei which feels like a lot.

For methods that use `computeInvariant` it makes more sense because everything depends on a ratio, and we need a new balance array anyways. In this case all of that is an extra, and there's no known edge case that breaks with the current implementation on `main`.

In other words, while this effectively pushes the rounding implicit fee even more in favor of the vault, it feels a bit too much given all the other rounding that we're applying. Moreover, these methods always charge a fee because they involve single tokens, so the taxable amount should always be positive for valid use cases (if it's not, chances are that `newBalances` will be below current balances and checked math will revert anyways). 

According to the snaps this ends up adding 110 bytes and ~1k gas for add single token exact out. I'd rather save that for another battle.